frontend/src/features/projects/common/components/ProjectFinancialTable.tsx

```diff
@@ -26,25 +27,29 @@ export const ProjectFinancialTable = ({ disabled, title }: IProjectFinacialTable
   const interestComponent = getIn(values, 'interestComponent');
   const salesCost = getIn(values, 'salesCost');
   const netBook = getIn(values, 'netBook');
-  React.useEffect(() => {
-    // Calculate the Gain before SPL.
-    setFieldValue(
-      'gainBeforeSpl',
-      calcGainBeforeSpl(
-        getNumber(market),
-        getNumber(interestComponent),
-        getNumber(salesCost),
-        getNumber(netBook),
-      ),
+  const gainBeforeSpl = getIn(values, 'gainBeforeSpl');
+  useEffect(() => {
+    const calculatedValue = calcGainBeforeSpl(
+      getNumber(market),
+      getNumber(interestComponent),
+      getNumber(salesCost),
+      getNumber(netBook),
     );
-  }, [market, interestComponent, salesCost, netBook, setFieldValue]);
+    if (gainBeforeSpl !== calculatedValue) {
```

Comment:
one of the few code changes in this PR, don't trigger a state update unless the value has actually changed.